backend/tilavarauspalvelu/api/graphql/types/reservation/serializers/staff_create_serializers.py

```diff
@@ -160,6 +162,16 @@ def create(self, validated_data: StaffCreateReservationData) -> Reservation:
             reservation: Reservation = super().create(validated_data)
             reservation.reservation_units.set([reservation_unit])
 
+        # After creating the reservation, check again if there are any overlapping reservations.
+        # This can fail if two reservations are created for reservation units in the same
+        # space-resource hierarchy at almost the same time, meaning when we check for overlapping
+        # reservations during validation, neither of the reservations are yet created.
+        overlapping_reservations = list(reservation.actions.overlapping_reservations().values_list("pk", flat=True))
+        if overlapping_reservations:
```

Comment:
Or simply just `reservation.actions.overlapping_reservations().exists()` ?

Reply:
Yes, previously had Sentry logging here but took it our as unnecessary.